test/framework/src/main/java/org/elasticsearch/test/AbstractSchemaValidationTestCase.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.AdditionalPropertiesValidator;
+import com.networknt.schema.ItemsValidator;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.JsonValidator;
+import com.networknt.schema.PropertiesValidator;
+import com.networknt.schema.SchemaValidatorsConfig;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.PathUtils;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+
+/**
+ * Test case for validating {@link ToXContent} objects against a json schema.
+ */
+public abstract class AbstractSchemaValidationTestCase<T extends ToXContent> extends ESTestCase {
+
+    public final void testSchema() throws IOException {
+        BytesReference xContent = XContentHelper.toXContent(createTestInstance(), XContentType.JSON, getToXContentParams(), false);
```

Comment:
This method tests one instance per run.  Randomization will eventually catch all the errors, but maybe not until after a PR is merged.  Some of the serialization tests test 10 random instances in a loop on each run, to increase the chance of finding the buggy branch on a single build.  Doing that here would reduce the risk of mistakes getting past CI and into the periodic builds where they'll cause noise for everyone.

As far as I can see, if this line were moved down closer to the point where `xContent` is first used then only 4 lines of this method would need to be in that loop.  Loading the schema could still just be done once.